homeassistant/exceptions.py

```diff
@@ -21,6 +21,16 @@ def __init__(
         translation_placeholders: dict[str, str] | None = None,
     ) -> None:
         """Initialize exception."""
+        # pylint: disable-next=import-outside-toplevel
+        from .helpers.translation import async_get_exception_message
+
+        if not args and (
+            message := async_get_exception_message(
```

Comment:
This doesn't feel like the right place to look up the translation, it seems like this should happen at the display layer

Reply:
> Ideally, we still have English in the log as it makes it very hard to debug code when all the errors are in another language, and all the code is in English

The generated message is in English.

Reply:
> This doesn't feel like the right place to look up the translation, it seems like this should happen closer to the display layer where the exception is consumed.

Right, this could happen (once per object) in an overloaded `__str__` method instead.